collector/agent_poller.go

```diff
@@ -69,80 +61,120 @@ type AgentState struct {
 	executorAppNames map[string]string
 }
 
+type Agent struct {
+	AgentIP        string
+	IPCommand      string
+	Port           int
+	PollPeriod     int
+	Topic          string
+	AgentStateChan chan<- *AgentState
+}
+
+func NewAgent(
+	ipCommand string,
+	port int,
+	pollPeriod int,
+	topic string) (Agent, error) {
+	a := Agent{}
+	if len(ipCommand) < 0 {
+		return a, errors.New("Must pass ipAddress to NewAgent()")
+	}
+	if port < 1024 {
+		return a, errors.New("Must pass port to NewAgent()")
+	}
+	if pollPeriod == 0 {
+		return a, errors.New("Must pass pollPeriod to NewAgent()")
+	}
+	if len(topic) < 1 {
```

Comment:
this also looks weird
if you want to check if the string is empty you can do
`if topic == ""`
or
`if len(topic) == 0`
i prefer first, because it explicitly tell that you are comparing a string not a slice